File: 3577. Count the Number of Computer Unlocking Permutations.cpp
class Solution {
public:
    int MOD = 1'000'000'007;

    int countPermutations(vector<int>& complexity) {
        long long ans = 1;
        int first = complexity[0];
        bool passedFirst = false;

        for (int comp : complexity) {
            if ((comp == first || comp < first) && passedFirst) return 0;
            passedFirst = true;
        }

        for (int i = 2; i < (int)complexity.size(); i++) {
            ans = (ans * i) % MOD;
        }

        return (int)ans;
    }
};

File: 3577. Count the Number of Computer Unlocking Permutations.java
class Solution {
    public int MOD = 1_000_000_007;
    public int countPermutations(int[] complexity) {
        long ans = 1;
        int first = complexity[0];
        boolean passedFirst = false;
        for(int comp : complexity){
            if((comp == first || comp < first) && passedFirst) return 0;
            passedFirst = true;
        }

        for(int i = 2; i < complexity.length; i++){
            ans = (ans * i) % MOD;
        }
        return (int)ans;
    }
}